### DISCUSSION TITLE
Inquiry regarding step estimation in JUDI.

### DISCUSSION THREAD

**Discussion by zhangxiaoshuotttt** [slimgroup/JUDI.jl#238] [Q&A] [answered]
@mloubout  @ziyiyin97 Hello, 
I'm interested in locating the codes related to step estimation in JUDI. 

Despite my attempts to search for relevant keywords, I haven't found much information on JUDI's step estimation method. While many examples in JUDI focus on the search gradient direction, there is little mention of the algorithm or principle utilized for step size estimation. 
I am curious whether step estimation method is a critical factor in FWI. Nonetheless, I am still curious and would like to test different methods. 

Thank you kindly for any assistance you can provide.

**Comment by ziyiyin97** [comment]
Some examples use a line search algorithm to determine the step length, e.g. https://github.com/slimgroup/JUDI.jl/blob/395793c9fd98f48922926f82cc2ab1d2a3bc3ff0/examples/scripts/fwi_example_2D.jl#L63 but I will let @mloubout provide further comments.

**Reply by zhangxiaoshuotttt** [reply]
Thanks a lot, and I'll read it carefully.

**Comment by mloubout** [comment, marked as answer]
> I haven't found much information on JUDI's step estimation method

JUDI itself does not estimate a step or define optimization algorithms so this is expected. In principle, JUDI is a framework that allows you to easily define the linear operators (forward, Jacobian) so that you can define your own problem in a simple and mathematical way.

In practice, for FWI, you will be some sort of step estimation as unit step length tend to be too simplistic to lead to useful results. To this end we show a few use case in the examples.Thos examples are all setup to use existing mathematical optimization libraries (NlOpt, SlimOptim, ...) or a simple steepest descent using the LineSearch.jl package to compute the step length (the example linked above).

The way JUDI is designed,  you should be able to easily either use the optimization package of your choice or implement a step estimation from the literature.

**Reply by zhangxiaoshuotttt** [reply]
Thanks for your further explanation. After reading some articles mentioning step estimation, I am curious about how much these algorithms can improve FWI. Based on your suggestions, I will proceed with the next test.
